pkgs/applications/networking/p2p/deluge/default.nix

```diff
@@ -1,39 +1,47 @@
-{ stdenv, fetchurl, fetchpatch, intltool, libtorrentRasterbar, pythonPackages }:
+{ stdenv, lib, fetchgit, intltool, python3Packages
+, libtorrentRasterbar
+, withGui ? true, gobject-introspection ? null, gtk3 ? null
+, libappindicator-gtk3 ? null, librsvg ? null, libnotify ? null, wrapGAppsHook ? null
+}:
 
-pythonPackages.buildPythonPackage rec {
-  name = "deluge-${version}";
-  version = "1.3.15";
+python3Packages.buildPythonApplication rec {
+  pname = "deluge";
+  version = "2.0.3";
 
-  src = fetchurl {
-    url = "http://download.deluge-torrent.org/source/${name}.tar.bz2";
-    sha256 = "1467b9hmgw59gf398mhbf40ggaka948yz3afh6022v753c9j7y6w";
+  src = fetchgit {
+    url = "https://git.deluge-torrent.org/deluge/";
+    sha256 = "0zav045bp3z3hrp6jni6wr9ryihgf1ngvgrn9vgbbwrhgdf3d3yg";
+    rev = "deluge-${version}";
   };
 
-  patches = [
-    # Fix preferences when built against libtorrent >=0.16
-    (fetchpatch {
-      url = "https://git.deluge-torrent.org/deluge/patch/?id=38d7b7cdfde3c50d6263602ffb03af92fcbfa52e";
-      sha256 = "0la3i0lkj6yv4725h4kbd07mhfwcb34w7prjl9gxg12q7px6c31d";
-    })
-  ];
+  postPatch = ''
+    echo ${version} > RELEASE-VERSION
+  '';
+
+  buildInputs = lib.optionals withGui [ gobject-introspection gtk3 libnotify librsvg ];
+
+  propagatedBuildInputs = with python3Packages; [
+    dbus-python geoip2 pillow rencode setproctitle six
+    twisted Mako chardet pyxdg pyopenssl service-identity
+    (libtorrentRasterbar.override { inherit (python3Packages) python; }).python
```

Comment:
Why is this done. `libtorrentRasterbar` exists in `python-packages.nix` and should be build for your interpreter already.